docs/standard/analyzers/platform-compat-analyzer.md

```diff
@@ -0,0 +1,366 @@
+
+      // warns: 'SupportedOnWindowsAndLinuxOnly' is supported on 'Windows'
+      // warns: 'SupportedOnWindowsAndLinuxOnly' is supported on 'Linux'
+      SupportedOnWindowsAndLinuxOnly();
+
+      // warns: 'ApiSupportedFromWindows8UnsupportFromWindows10' is supported on 'windows' 8.0 and later  
+      // warns: 'ApiSupportedFromWindows8UnsupportFromWindows10' is unsupported on 'windows' 10.0.19041.0 and later
+      ApiSupportedFromWindows8UnsupportFromWindows10();
+
+      // warns: 'ApiSupportedFromWindows8UnsupportFromWindows10' is supported on 'windows' 10.0.19041.0 and later
+      // for same platform analyzer only warn for the latest version.
+      AssemblySupportedOnWindowsApiSupportedFromWindows10();
+  }
+
+  // an API not supported on android but supported on all other.
+  [UnsupportedOSPlatform("android")]  
+  public void DoesNotWorkOnAndroid() { }
+
+  // an API was unsupported on Windows until version 8.0.
+  // The API is considered supported everywhere else without constraints.
+  [UnsupportedOSPlatform("windows")]
+  [SupportedOSPlatform("windows8.0")]
+  public void StartedWindowsSupportFromVersion8() { }
+
+  // an API was unsupported on Windows until version8.0.
+  // Then the API is removed (unsupported) from version 10.0.19041.0.
+  // The API is considered supported everywhere else without constraints.
+  [UnsupportedOSPlatform("windows")]
+  [SupportedOSPlatform("windows8.0")]
+  [UnsupportedOSPlatform("windows10.0.19041.0")]
+  public void StartedWindowsSupportFrom8UnsupportedFrom10() { }
+
+  public void Caller2()
+  {
+      DoesNotWorkOnAndroid(); // warns 'DoesNotWorkOnAndroid' is unsupported on 'android'
+
+      // warns:'StartedWindowsSupportFromVersion8' is unsupported on 'windows'  
+      // warns:'StartedWindowsSupportFromVersion8' is supported on 'windows' 8.0 and later
+      StartedWindowsSupportFromVersion8();
+
+      // warns:'StartedWindowsSupportFrom8UnsupportedFrom10' is unsupported on 'windows'  
+      // warns:'StartedWindowsSupportFrom8UnsupportedFrom10' is supported on 'windows' 8.0 and later
+      // even there were 3 diagnostics found analyzer warn only for the first 2.
+      StartedWindowsSupportFrom8UnsupportedFrom10();
+  }
+  ```
+
+## Handle reported warnings
+
+The recommended way to deal with these diagnostics is to make sure you only call platform-specific APIs when running on an appropriate platform. Following are the options you can use to address the warnings; choose whichever is most appropriate for your situation:
+
+- **Guard the call**. You can either achieve this by excluding the code at build time using `#if` and multi-targeting or by conditionally calling the code at run time. Check whether you’re running on a desired `Platform` by using one of platform-check methods, for example, `OperatingSystem.Is<Platform>()` or `OperatingSystem.Is<Platform>VersionAtLeast(int major, int minor = 0, int build = 0, int revision = 0)`.
```

Comment:
We should separate the `#if` and `OperatingSystem` approaches. They're quite distinct. And I recommend we move the `#if` approach to the end, as it requires defining custom MSBuild constants per target.